src/web/mod.rs

```diff
@@ -466,6 +466,24 @@ fn redirect(url: Url) -> Response {
     resp
 }
 
+pub fn redirect_base(req: &Request) -> String {
+    // Try to get the scheme from CloudFront first, and then from iron
+    let scheme = req.headers
+        .get_raw("cloudfront-forwarded-proto")
+        .and_then(|values| values.get(0))
+        .and_then(|value| std::str::from_utf8(value).ok())
+        .filter(|proto| *proto == "http" || *proto == "https")
+        .unwrap_or_else(|| req.url.scheme());
+
+    // Only include the port if it's needed
+    let port = req.url.port();
+    if port == 80 {
```

Comment:
Shouldn't this be 80 or 443?

Reply:
Uh, probably yes, but it shouldn't cause any issue with our production setup, so fixing it is not that high priority.

Reply:
I think only 80 is fine since CloudFront is sending requests to 80.